spec/controllers/questionnaires_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+require 'rails_helper'
+
+# Execute all tests in the context of the QuestionnairesController
+describe QuestionnairesController do
+
+  # Perform tests on the validate quiz method of the
+  # questionnaires controller.
+  describe 'Validate Quiz' do
+
+    # Create a new controller instance for testing.
+    before :each do
+
+      # Create an instructor
+      @instructor = create(:instructor)
+
+      #Create an assignment with quiz
+      @assignment = create :assignment, require_quiz: true, instructor: @instructor, course: nil, num_quiz_questions: 1
+
+      # Create a participant and assignment team
+      @submitter = create :participant, assignment: @assignment
+      @team = create :assignment_team, assignment: @assignment
+      create :team_user, team: @team, user: @submitter.user
+
+      # Define needed hash methods to emulate params.
+      class Hash
+        def require key
+          return self[key]
+        end
+        def permit *args
+          return self
+        end
+      end
+
+      # Create a params stub for the controller to work with
+      @params = {
+          :questionnaire => {
+              :name => 'Quiz for test',
+              :id => '',
+              :type => 'QuizQuestionnaire'
+          },
+          :new_question => {
+              '1' => 'Test Question 1'
+          },
+          :new_choices => {
+              '1' => {
+                  'MultipleChoiceRadio' => {
+                      '1' => {
+                          :iscorrect => '1',
+                          :txt => 'Test Quiz 1'
+                      },
+                      '2' => {
+                          :txt=> 'Test Quiz 2'
+                      },
+                      '3' => {
+                          :txt=>'Test Quiz 3'
+                      },
+                      '4' => {
+                          :txt => 'Test Quiz 4'
+                      }
+                  },
+                  'MultipleChoiceCheckbox' => {
+                      '1' => {
+                          :iscorrect => '0',
+                          :txt => ''
+                      },
+                      '2' => {
+                          :iscorrect => '0',
+                          :txt => ''
+                      },
+                      '3' => {
+                          :iscorrect => '0',
+                          :txt => ''
+                      },
+                      '4' => {
+                          :iscorrect => '0',
+                          :txt => ''
+                      }
+                  }
+              }
+          },
```

Comment:
The @params above does not look correct to me. There is only one question, it cannot be multiple choice and multiple select at the same time.

Here is an example of params with 2 questions, one multiple choice and one multiple select:

{"utf8"=>"✓",
 "authenticity_token"=>
  "cCL0nD0p1Icv9aZUGu2eWMWDLKW7XrBjcyou2iHLgKN7IJ1YzLbMB6D4tk4o6XH2SYHUEqsjuhdUQBOvByGwJA==",
 "questionnaire"=>
  {"name"=>"Quiz for test", "id"=>"", "type"=>"QuizQuestionnaire"},
 "new_question"=>{"1"=>"Test Question 1", "2"=>"Test Question 2"},
 "new_choices"=>
  {"1"=>
    {"MultipleChoiceRadio"=>
      {"1"=>{"iscorrect"=>"1", "txt"=>"Test Quiz 1"},
       "2"=>{"txt"=>"Test Quiz 2"},
       "3"=>{"txt"=>"Test Quiz 3"},
       "4"=>{"txt"=>"Test Quiz 4"}},
     "MultipleChoiceCheckbox"=>
      {"1"=>{"iscorrect"=>"0", "txt"=>""},
       "2"=>{"iscorrect"=>"0", "txt"=>""},
       "3"=>{"iscorrect"=>"0", "txt"=>""},
       "4"=>{"iscorrect"=>"0", "txt"=>""}}},
   "2"=>
    {"MultipleChoiceRadio"=>
      {"1"=>{"txt"=>""}, "2"=>{"txt"=>""}, "3"=>{"txt"=>""}, "4"=>{"txt"=>""}},
     "MultipleChoiceCheckbox"=>
      {"1"=>{"iscorrect"=>"1", "txt"=>"Test Quiz 5"},
       "2"=>{"iscorrect"=>"0", "txt"=>"Test Quiz 6"},
       "3"=>{"iscorrect"=>"0", "txt"=>"Test Quiz 7"},
       "4"=>{"iscorrect"=>"1", "txt"=>"Test Quiz 8"}}}},
 "question_type"=>
  {"1"=>{"type"=>"MultipleChoiceRadio"},
   "2"=>{"type"=>"MultipleChoiceCheckbox"}},
 "save"=>"Create Quiz",
 "aid"=>"776",
 "pid"=>"32085",
 "controller"=>"questionnaires",
 "action"=>"create_quiz_questionnaire"}